File: src/main/java/ir/reservs/reservs/ui/main/history/HistoryContract.java
package ir.reservs.reservs.ui.main.history;

import java.util.List;

import ir.reservs.reservs.model.ReserveHistory;
import ir.reservs.reservs.ui.base.BaseFragmentContract;

public class HistoryContract {

    public interface View extends BaseFragmentContract.View {
        void setHistoryData(List<ReserveHistory> reserves);

        void loadingState();

        void normalState();

        void errorState();

        void emptyState();
    }

    public interface Presenter extends BaseFragmentContract.Presenter<View> {
        void getDataHistory();
    }


}

File: src/main/java/ir/reservs/reservs/ui/main/history/HistoryContract.kt
package ir.reservs.reservs.ui.main.history

import androidx.lifecycle.LifecycleOwner
import ir.reservs.reservs.model.History
import ir.reservs.reservs.ui.base.BaseFragmentContract

class HistoryContract {

    interface View : BaseFragmentContract.View, LifecycleOwner {
        fun updateAdapter(it: MutableList<History>)
    }

    interface Presenter : BaseFragmentContract.Presenter<View>


}